Hide headers per default; Add flag `-i` for showing headers

makkes opened the pull request:
This changes the default behaviour of this plugin in that now HTTP headers are mot peintedper default. This is very useful when you want to process the output with `jq` for instance. Also, it aligns more with `curl`s behaviour. The `-i`/`--include` flag shows headers.

Commit:
Add flag `--suppress-headers` for suppressing header output

makkes commented:
@bamarni what do you think?

bamarni commented:
That's definitely needed :+1: 

As the tool is mimicking the `curl` CLI for flags in general, can we follow the same approach here? That is, by default no headers are printed unless the verbose option is present. From the plugin this can be checked through the `DCOS_VERBOSITY` env variable, which is passed when invoking the plugin:

- `dcos -v http [...]` (`DCOS_VERBOSITY=1`)
- `dcos -vv http [...]` (`DCOS_VERBOSITY=2`)

In these both cases only we would print headers.

bamarni commented:
Just merged https://github.com/dcos/dcos-http-cli/pull/4 related to headers as well, sorry for the conflict!

makkes commented:
Thanks for the feedback, @barmani. I actually didn't want to break the default behaviour. But if we can live with that, even better.

If we wanted to have 100% curl compatibility we could also just use `-i`/`--include` and reserve `-v` for even more verbosity.

bamarni commented:
Yeah that's right, `-v` is for request headers, to control response headers it should be `-i, --include` indeed.

This is mostly a plugin used as a reference for plugin authors, it's fine to have breaking changes.

Commit:
Flag -i/--include now triggers inclusion of HTTP headers in output

The default is now to not print the headers. This complies with the
default behaviour of curl.

makkes commented:
Ok, `-i`/`--include` now includes the HTTP headers in the output. The default is to not print them.
